src/base.ts

```diff
@@ -132,36 +124,60 @@ export const fetchPath = async (
   }
 }
 
-export const postData = async (
-  service: Service,
-  path: string,
+export const postData = async ({
+  path,
+  body,
+  options,
+}: {
+  path: string
   // eslint-disable-next-line @typescript-eslint/ban-types
-  body: object,
-  headers = {}
-) => {
-  return fetchPath(service, path, {
-    method: "POST",
+  body: object
+  options: ApiCallOptions
```

Comment:
I might miss something but isn't the base file providing internal interfaces so it should be consistent for all helpers provided by this file?

Reply:
Jain. `ApiCallOptions` doesn't specify method and we don't want people setting method to `GET` when invoking `postData`

Reply:
yeah I see the pint on the method here but the rest should be consisten like if you want to have the generic fetch options for example (for debugging) I would expect to be able to set them in a post or put request the same way...

Reply:
That's why `RequestOptions` extends `ApiCallOptions` with the method only.
We have a `fetchPath` that's the lowest level wrapper around fetch and we have `postData` and fiends that are wrappers around `fetchPath`. We don't have a wrapper around the `get` request - is that what you're hinting at?